File: ios/RNTCamera/RNTCameraViewManager.m
//
//  RNTCameraViewManager.m
//  MerantixCam
//
//  Created by Malith Thilakarathne on 2023-01-01.
//

#import <Foundation/Foundation.h> // TODO: Remove this if not needed
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(RNTCameraViewManager, RCTViewManager)
RCT_EXPORT_VIEW_PROPERTY(onCapture, RCTBubblingEventBlock)
RCT_EXTERN_METHOD(takePhoto:(nonnull NSNumber *)node)
@end

File: ios/RNTCamera/RNTCameraViewManager.swift
//
//  RNTCameraViewManager.swift
//  MerantixCam
//
//  Created by Malith Thilakarathne on 2023-01-01.
//

import Foundation

@objc(RNTCameraViewManager)
class RNTCameraViewManager: RCTViewManager {
  override func view() -> UIView! {
    return RNTCameraView()
  }

	override static func requiresMainQueueSetup() -> Bool {
    return true
  }

  @objc func takePhoto(_ node: NSNumber) {
      DispatchQueue.main.async {
        let component = self.bridge.uiManager.view(
            forReactTag: node
        ) as! RNTCameraView
        component.takePhoto()
      }
  }
}